### PR TITLE
bin: don't create db if missing when dump

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang @disksing @zhangjinpeng1987 PTAL

### CODE REVIEW

**Comment by disksing** [comment]
LGTM.

File: src/util/rocksdb.rs [hunk 1]
```diff
@@ -19,6 +19,18 @@ pub fn get_cf_handle<'a>(db: &'a DB, cf: &str) -> Result<&'a DBCFHandle, String>
         .ok_or_else(|| format!("cf {} not found.", cf))
 }
 
+pub fn open_engine(path: &str, cfs: &[&str]) -> Result<DB, String> {
```
Comment (ngaut) [inline src/util/rocksdb.rs]: s/open_engine/open/

File: src/util/rocksdb.rs [hunk 2]
```diff
@@ -19,6 +19,18 @@ pub fn get_cf_handle<'a>(db: &'a DB, cf: &str) -> Result<&'a DBCFHandle, String>
...
+    let mut opts = Options::new();
+    opts.create_if_missing(false);
+    open_engine_opt(&opts, path, cfs)
+}
+
+pub fn open_engine_opt(opts: &Options, path: &str, cfs: &[&str]) -> Result<DB, String> {
```
Comment (ngaut) [inline src/util/rocksdb.rs]: s/open_engine_opt/open_opt/

**Comment by ngaut** [comment]
LGTM

**Pull request merged**